src/seeds.gratitude.cpp

```diff
@@ -70,61 +99,173 @@ ACTION gratitude::acknowledge (name from, name to, string memo) {
         item.receivers.push_back(to);
     });
   }
+
+  // Updates ack stats
+  auto stitr = stats2.rbegin();
+  auto round_id = stitr->round_id;
+
+  auto stitr2 = stats2.find(round_id);
+  auto oldacks = stitr2->num_acks;
+  stats2.modify(stitr2, _self, [&](auto& item) {
+      item.num_acks = oldacks + 1;
+  });
+
 }
 
+
 ACTION gratitude::testacks() {
   require_auth(get_self());
 
   auto actr = acks.begin();
+
   while (actr != acks.end()) {
-    calc_acks(actr->donor);
+    _calc_acks(actr->donor);
     actr = acks.erase(actr);
   }
 }
 
-ACTION gratitude::newround() {
+ACTION gratitude::calcacks(uint64_t start) {
   require_auth(get_self());
 
-  auto contract_balance = eosio::token::get_balance(contracts::token, get_self(), seeds_symbol.code());
-  uint64_t tot_accounts = get_size("balances.sz"_n);
-  uint64_t volume = get_current_volume();
+  auto actr = start == 0 ? acks.begin() : acks.find(start);
 
-  auto actr = acks.begin();
-  while (actr != acks.end()) {
-    calc_acks(actr->donor);
+  if (actr != acks.end()) {
+    _calc_acks(actr->donor);
     actr = acks.erase(actr);
   }
+  // If there is still more, do recursion call
```

Comment:
did you test that the recursive calling works? for example, has the permission, eosio.code on active, etc, 

I usually give it a batchsize parameter so that I can test it with batch size 1 in the unit tests

But if you tested it, cool, just checking and making sure.

Reply:
Yes, tested.